### PR TITLE
Added CSS documentation for datagrid and modal components

### PR DESCRIPTION
**Pull request by vishnuchoutapellys**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/data/datagrid/API.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Datagrid - Style API
+
+## CSS Variables
+
+| CSS Variable Name                                     | Description                                                                  |
+| ----------------------------------------------------- | -----------------------------------------------------------------------------|
+| --clr-datagrid-font-color                             | Changes the text color of datagrid.                                          |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:7]: Changes the datagrid text color

File: projects/angular/src/data/datagrid/API.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-default-border-color                   | Changes the border color of datagrid when default.                           |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:8]: Changes the default datagrid border color

File: projects/angular/src/data/datagrid/API.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-icon-color                             | Changes the header icons color of datagrid.                                  |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:9]: Changes the datagrid icons color

// there is nothing about header here, it's used for the action button, detail button and expandable row caret button

File: projects/angular/src/data/datagrid/API.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-row-hover-color                        | Changes the row background color of the data table when hovers over element. |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:10]: Changes the datagrid row color when hovered over.

File: projects/angular/src/data/datagrid/API.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-row-hover-font-color                   | Changes the row text color of data table when hovers over element.           |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:11]: Changes the datagrid row text color when hovered over.

File: projects/angular/src/data/datagrid/API.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-action-toggle-color                    | Changes the toggle action button color.                                      |
+| --clr-datagrid-pagination-btn-color                   | Changes the table bottom pagination buttons color.                           |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:13]: Changes the datagrid pagination buttons color

File: projects/angular/src/data/datagrid/API.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-pagination-btn-disabled-color          | Changes the table bottom pagination buttons color when disabled.             |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:14]: Pls remove all "table" and "bottom" occurrences in the file.
We have only one pagination, so it's not necessary to specify that it's bottom. Also it's just "datagrid". Not "table" or "table datagrid".

File: projects/angular/src/data/datagrid/API.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-pagination-input-border-color          | Changes the text box border color of table datagrid.                         |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:15]: Changes the datagrid pagination input border color

File: projects/angular/src/data/datagrid/API.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-pagination-input-border-focus-color    | Changes the text box border color of table datagrid when focus on element.   |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:16]: Changes the datagrid pagination input border color when focused.

File: projects/angular/src/data/datagrid/API.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-popover-bg-color                       | Changes the background color of popup box.                                   |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:17]: Changes the datagrid popovers background color

// this includes filters, action menus, column management popup

File: projects/angular/src/data/datagrid/API.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-popover-border-color                   | Changes the border color of popup box.                                       |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:18]: Changes the datagrid popovers border color.

File: projects/angular/src/data/datagrid/API.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-row-selected                           | Changes the text color of table row when user selected.                      |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:19]: Changes the datagrid selected row text color

File: projects/angular/src/data/datagrid/API.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-loading-background                     | Changes the background color of datagrid when click on toggle spinner.       |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:20]: Changes the datagrid loading spinner background

// you don't click on the spinner. it get programmatically enabled/disabled.

File: projects/angular/src/data/datagrid/API.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-popovers-box-shadow-color              | Changes the popup box shadow color of datagrid.                              |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:21]: Changes the datagrid popovers shadow color.

File: projects/angular/src/data/datagrid/API.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-switch-header-font-color        | Changes the column header text color of datagrid.                            |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:22]: Changes the datagrid manage columns dialog header font color.

// this was a hard one :) needed to dig in the code then debug the component to see what it does.

File: projects/angular/src/data/datagrid/API.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-switch-header-font-hover-color  | Changes the column header text color of datagrid when hovers over element.   |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:23]: Changes the datagrid manage columns dialog header font color when hovered.

File: projects/angular/src/data/datagrid/API.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-detail-caret-icon-open-bg-color        | Changes the caret icon background color of datagrid.                         |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:24]: Changes the datagrid open detail caret icon background color.

File: projects/angular/src/data/datagrid/API.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-detail-caret-icon-open-icon-color      | Changes the caret icon color of datagrid.                                    |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:25]: Changes the datagrid open detail caret icon color.

File: projects/angular/src/data/datagrid/API.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-placeholder-color                      | Changes the text color of datagrid placeholder.                              |
+| --clr-datagrid-placeholder-font-size                  | Changes the placeholder text size of datagrid.                               |
+| --clr-datagrid-pagination-btn-disabled-opacity        | Changes the opacity of pagination button when disabled.                      |
+| --clr-datagrid-column-toggle-border-color             | Changes the toggle button border color.                                      |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:29]: Changes the **column** toggle button border color

File: projects/angular/src/data/datagrid/API.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-toggle-fill-color               | Changes the background color of toggle button.                               |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:30]: Same here and below - don't skip "column". Please change to "column toggle" in all occurrences. 
It's about a feature that lets the user show and hide columns.

File: projects/angular/src/data/datagrid/API.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-toggle-text-color               | Changes the text color of pagination toggle button.                          |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:31]: There is nothing about pagination here. We're still on the "column toggle" feature.

File: projects/angular/src/data/datagrid/API.md [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-toggle-border-hover-color       | Changes the border color of toggle button when hovers over element.          |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:32]: See previous comments. Finish with "...when hovered over."

File: projects/angular/src/data/datagrid/API.md [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-toggle-fill-hover-color         | Changes the background color of toggle button when hovers over element.      |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:33]: Same

File: projects/angular/src/data/datagrid/API.md [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-toggle-text-hover-color         | Changes the text color of toggle button when hovers over element.            |
+| --clr-datagrid-column-toggle-border-active-color      | Changes the border color of toggle button when active.                       |
+| --clr-datagrid-column-toggle-fill-active-color        | Changes the background color of toggle button when active.                   |
+| --clr-datagrid-column-toggle-text-active-color        | Changes the text color of pagination toggle button when active.              |
+
+
+## Class names
+
+| Class name                           | Description                                                                     |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:42]: Some of the following are too obvious, like "clr-datagrid" and just identify the main datagrid element. Some other are deeply implementation specific and I see no clear way of describing them in a "reference" format. Like the ".datagrid-calculation-table".
I have no idea how they should be documented, besides the main ones like the "clr-datagrid".

**Review by Jinnie — COMMENTED** [review]
Vishnu and Manjari, thanks for the detailed work. I realize a lot of this is hard to properly describe without more context and deeper understanding of the code.

Some do not have better description than the name of the id itself.
Some require deep digging and debugging to verify what they are doing.
On the classes part, we're trying to document deep implementation details that can be changed and become obsolete at any moment.
@williamernest, I suggest we continue this work by talking a bit more about the goals, detail level, levels of API we want to expose.

File: projects/angular/src/data/datagrid/API.md (hunk 10 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:17]: As per your suggestions, changed

File: projects/angular/src/data/datagrid/API.md (hunk 11 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:18]: As per your suggestions, changed

File: projects/angular/src/data/datagrid/API.md (hunk 1 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: As per your suggestions, changed it.

File: projects/angular/src/data/datagrid/API.md (hunk 2 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: As per your suggestions, changed it.

File: projects/angular/src/data/datagrid/API.md (hunk 3 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Yes, changed it as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 4 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: As per your suggestions, changed it.

File: projects/angular/src/data/datagrid/API.md (hunk 5 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: As per your suggestions, changed it.

File: projects/angular/src/data/datagrid/API.md (hunk 20 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: As per your suggestions, added column text.

File: projects/angular/src/data/datagrid/API.md (hunk 21 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: As per your suggestions, changed it.

File: projects/angular/src/data/datagrid/API.md (hunk 22 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per previous comments.

File: projects/angular/src/data/datagrid/API.md (hunk 23 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: As per your previous comments, changed it.

File: projects/angular/src/data/datagrid/API.md (hunk 6 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 7 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Removed all table and bottom occurrences in the file.

File: projects/angular/src/data/datagrid/API.md (hunk 13 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 15 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 16 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 17 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 18 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 19 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 14 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 8 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestions.

File: projects/angular/src/data/datagrid/API.md (hunk 9 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestions.

File: projects/angular/src/data/datagrid/API.md (hunk 12 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md]: Changed as per your suggestions.

File: projects/angular/src/data/datagrid/API.md [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+| --clr-datagrid-column-toggle-fill-active-color        | Changes the background color of toggle button when active.                   |
+| --clr-datagrid-column-toggle-text-active-color        | Changes the text color of pagination toggle button when active.              |
+
+
+## Class names
+
+| Class name                            | Description                                                                     |
+| ------------------------------------- | -----------------------------------------------------------------------------   |
+| .clr-datagrid                         | Identify the main datagrid element.                                             |
+| .datagrid-nav                         | To filter and navigate through data displayed in datagrid.                      |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:45]: This is demo-app only class, please remove. It's not part of the library.

File: projects/angular/src/data/datagrid/API.md [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-outer-wrapper               | It's container element that holds and positions data grid component.            |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:46]: Datagrid internal layout container

File: projects/angular/src/data/datagrid/API.md [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-calculation-table           | Allows for the display and manipulation of calculated data.                     |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:47]: Datagrid internal element involved in automatic column size calculation

File: projects/angular/src/data/datagrid/API.md [hunk 28]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-host                        | To manage and optimize the performance of datagrid.                             |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:48]: Basic datagrid layout container

File: projects/angular/src/data/datagrid/API.md [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-header                      | To displays the column labels or headers.                                       |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:49]: Datagrid header container

File: projects/angular/src/data/datagrid/API.md [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-row                         | To display single record or item of data in a horizontal format.                |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:50]: Datagrid row container

File: projects/angular/src/data/datagrid/API.md [hunk 31]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-row-master                  | Primary row that used to group or summarize related data in hierarchical format.|
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:51]: Internal datagrid row element

File: projects/angular/src/data/datagrid/API.md [hunk 32]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-row-flex                    | Allows for flexible sizing and positioning of cells within data grid.           |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:52]: Internal datagrid row container, managing "flex" positioning

File: projects/angular/src/data/datagrid/API.md [hunk 33]
```diff
@@ -0,0 +1,53 @@
...
+| .datagrid-footer                      | The bottom row that typically displays summary.                                 |
```
Comment (Jinnie) [inline projects/angular/src/data/datagrid/API.md:53]: Just "Datagrid footer"

File: projects/angular/src/modal/API.md [hunk 34]
```diff
@@ -0,0 +1,26 @@
+# Modal - Style API
+
+## CSS Variables
+
+| CSS Variable Name                                     | Description                                                                  |
+| ----------------------------------------------------- | -----------------------------------------------------------------------------|
+| --clr-modal-close-color                               | Changes the close icon color.                                                |
+| --clr-modal-bg-color                                  | Changes the background color of popup modal.                                 |
```
Comment (Jinnie) [inline projects/angular/src/modal/API.md:8]: popup modal -> modal popup

File: projects/angular/src/data/datagrid/API.md (hunk 25 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:45]: Removed as per your suggestion.

File: projects/angular/src/data/datagrid/API.md (hunk 26 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:46]: Changes as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 27 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:47]: Changes as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 28 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:48]: Changes as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 29 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:49]: Changes as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 30 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:50]: Changes as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 31 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:51]: Changes as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 32 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:52]: Changed as you suggested.

File: projects/angular/src/data/datagrid/API.md (hunk 33 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/data/datagrid/API.md:53]: Changed as you suggested.

File: projects/angular/src/modal/API.md (hunk 34 above)
Comment (vishnuchoutapellys) [reply projects/angular/src/modal/API.md:8]: Changed as you suggested.

**Comment by dtsanevmw** [comment]
Done in: #755 , #727

**Pull request closed without merging**